### DISCUSSION TITLE
Cypress. How intercept requests by request body

### DISCUSSION THREAD

**Discussion by sadikovandrew** [cypress-io/cypress#20014] [Questions and Help] [answered]
Hi! Please tell me how I can intercept requests by request body

Multiple requests are being made on the page:

POST: /some_module/Handlers/Ajax.ashx
![image](https://user-images.githubusercontent.com/95144904/152118408-fdab095e-8939-4629-bfa4-6d8a01800b93.png)

But all requests have different request body
![image](https://user-images.githubusercontent.com/95144904/152118726-29651d17-7bfb-48ce-99bb-4183f715ec2c.png)

How to write cy.intercept(...) correctly to check each request separately?

I tried using regular expressions in the query block

cy.intercept({
      method: "POST",
      url: "/some_module/Handlers/Ajax.ashx",
      query: "method=some_method1*",
    }).as("PostAjax1");

cy.intercept({
      method: "POST",
      url: "/some_module/Handlers/Ajax.ashx",
      query: "method=some_method2*",
    }).as("PostAjax2");

But it doesn't work

![image](https://user-images.githubusercontent.com/95144904/152120093-f50edcc9-2a51-486b-b693-0428f7d53a2c.png)

Thank you in advance!

**Comment by TheColorRed** [comment, marked as answer]
I am not sure if this is what you are looking for, but you can write a handler that will handle custom actions for the request. This is a function that is passed in as the second parameter:

```javascript
cy.intercept({
  method: 'POST',
  /* Your other request parameters */    
}, handler);

function handler(req) {
  if(req.body.something_here) {
    // Do something
  }
}
```

**Reply by sadikovandrew** [reply]
Thanks! This alternative way works!
